Assets/Scripts/CharacterControllers/Melody/States/IdleState.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace Melody.States
+{
+    public class IdleState : MelodyState
+    {
+
+        public IdleState(MelodyController controller) : base(controller) { }
+
+        protected override void Enter() { }
+
+        public override void OnUpdate(float time)
+        {
+            base.OnUpdate(time);
+
+            if (melodyController.input.AttackButtonDown())
+            {
+
+                ableToExit = true;
+                nextState = new AttackState(melodyController);
+            }
+            else if (melodyController.input.ParryButtonDown())
+            {
+                ableToExit = true;
+                nextState = new CounterState(melodyController);
+            }
+            else if (melodyController.input.DodgeButtonDown())
+            {
+                ableToExit = true;
+                nextState = new DodgeState(melodyController);
+            }
+            else if (melodyController.input.GetHorizontalMovement() != 0 || melodyController.input.GetVerticalMovement() != 0)
+            {
+                ableToExit = true;
+                nextState = new MovingState(melodyController);
```

Comment:
This is a problem for further down the line, but wanted to bring it up so we're thinking about it during the design phase:

We'll need to account for Melody having different attack/parry animations depending on whether she's moving or not (I'd like for the player to still have largely reduced movement speed even while attacking/parrying). This could probably be handled within the attack/parry states by applying either the idle or walk animation to her legs while playing the attack/parry animation on her upper body using animation masks.

I'd consider this a problem for a future PR, though.

Reply:
We still might want to add a dampened movement option into the attack and parry states, even if we don't do anything differently with the animations yet. Let's say something like 1/3 the player's normal movement speed, but no rotation.